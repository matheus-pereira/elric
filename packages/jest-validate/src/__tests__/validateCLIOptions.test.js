/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import validateCLIOptions from '../validateCLIOptions';

test('validates yargs special options', () => {
  const options = ['$0', '_', 'help', 'h'];
  const argv = {
    $0: true,
    _: true,
    h: false,
    help: false,
  };
  expect(validateCLIOptions(argv, options)).toBe(true);
});

test('validates testURL', () => {
  const options = {
    testURL: {
      description: 'This option sets the URL for the jsdom environment.',
      type: 'string',
    },
  };
  const argv = {
    testURL: 'http://localhost',
  };
  expect(validateCLIOptions(argv, options)).toBe(true);
});

test('fails for unknown option', () => {
  const options = ['$0', '_', 'help', 'h'];
  const argv = {
    $0: true,
    unknown: 'unknown',
  };
  expect(() =>
    validateCLIOptions(argv, options),
  ).toThrowErrorMatchingSnapshot();
});

test('fails for multiple unknown options', () => {
  const options = ['$0', '_', 'help', 'h'];
  const argv = {
    $0: true,
    elric: 'cool',
    test: 'unknown',
  };
  expect(() =>
    validateCLIOptions(argv, options),
  ).toThrowErrorMatchingSnapshot();
});

test('does not show suggestion when unrecognized cli param length <= 1', () => {
  const options = ['$0', '_', 'help', 'h'];
  const argv = {
    $0: true,
    l: true,
  };
  expect(() =>
    validateCLIOptions(argv, options),
  ).toThrowErrorMatchingSnapshot();
});

test('shows suggestion when unrecognized cli param length > 1', () => {
  const options = ['$0', '_', 'help', 'h'];
  const argv = {
    $0: true,
    hell: true,
  };
  expect(() =>
    validateCLIOptions(argv, options),
  ).toThrowErrorMatchingSnapshot();
});
