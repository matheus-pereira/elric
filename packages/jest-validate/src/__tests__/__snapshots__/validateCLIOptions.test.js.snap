// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`does not show suggestion when unrecognized cli param length <= 1 1`] = `
"<red><bold><bold>●</><bold> Unrecognized CLI Parameter</>:</>
<red></>
<red>  Unrecognized option <bold>\\"l\\"</>.</>
<red></>
<red>  <bold>CLI Options Documentation</>:</>
<red>  https://elricjs.io/docs/cli</>
<red></>"
`;

exports[`fails for multiple unknown options 1`] = `
"<red><bold><bold>●</><bold> Unrecognized CLI Parameters</>:</>
<red></>
<red>  Following options were not recognized:</>
<red>  <bold>[\\"elric\\", \\"test\\"]</></>
<red></>
<red>  <bold>CLI Options Documentation</>:</>
<red>  https://elricjs.io/docs/cli</>
<red></>"
`;

exports[`fails for unknown option 1`] = `
"<red><bold><bold>●</><bold> Unrecognized CLI Parameter</>:</>
<red></>
<red>  Unrecognized option <bold>\\"unknown\\"</>.</>
<red></>
<red>  <bold>CLI Options Documentation</>:</>
<red>  https://elricjs.io/docs/cli</>
<red></>"
`;

exports[`shows suggestion when unrecognized cli param length > 1 1`] = `
"<red><bold><bold>●</><bold> Unrecognized CLI Parameter</>:</>
<red></>
<red>  Unrecognized option <bold>\\"hell\\"</>. Did you mean <bold>\\"help\\"</>?</>
<red></>
<red>  <bold>CLI Options Documentation</>:</>
<red>  https://elricjs.io/docs/cli</>
<red></>"
`;
